templates/_helpers.tpl

```diff
@@ -31,6 +31,17 @@ Create chart name and version as used by the chart label.
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
+{{/*
+Create the name of the service account
+*/}}
+{{- define "temporal.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create -}}
+    {{ default (include "temporal.fullname" .) .Values.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.serviceAccount.name }}
```

Comment:
in some kubernetes environments there will not be 'default' service account at all. So it is better to skip 'else' section

Reply:
@nvtkaszpir  done